test/python/pulse/test_builder.py

```diff
@@ -0,0 +1,969 @@
+from qiskit.pulse.instructions import directives
+from qiskit.test import QiskitTestCase
+from qiskit.test.mock import FakeOpenPulse2Q
+from qiskit.pulse import pulse_lib, instructions
+
+# pylint: disable=invalid-name
+
+
+class TestBuilder(QiskitTestCase):
+    """Test the pulse builder context."""
+
+    def setUp(self):
+        self.backend = FakeOpenPulse2Q()
+        self.configuration = self.backend.configuration()
+        self.defaults = self.backend.defaults()
+        self.inst_map = self.defaults.instruction_schedule_map
+
+
+class TestContexts(TestBuilder):
+    """Test builder contexts."""
+    def test_align_sequential(self):
+        """Test the sequential alignment context."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+
+        with pulse.build() as schedule:
+            with pulse.align_sequential():
+                pulse.delay(3, d0)
+                pulse.delay(5, d1)
+                pulse.delay(7, d0)
+
+        reference = pulse.Schedule()
+        # d0
+        reference.insert(0, instructions.Delay(3, d0), mutate=True)
+        reference.insert(8, instructions.Delay(7, d0), mutate=True)
+        # d1
+        reference.insert(3, instructions.Delay(5, d1), mutate=True)
+
+        self.assertEqual(schedule, reference)
+
+    def test_align_left(self):
+        """Test the left alignment context."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+        d2 = pulse.DriveChannel(2)
+
+        with pulse.build() as schedule:
+            with pulse.align_left():
+                pulse.delay(11, d2)
+                pulse.delay(3, d0)
+                with pulse.align_left():
+                    pulse.delay(5, d1)
+                    pulse.delay(7, d0)
+
+        reference = pulse.Schedule()
+        # d0
+        reference += instructions.Delay(3, d0)
+        reference += instructions.Delay(7, d0)
+        # d1
+        reference = reference.insert(3, instructions.Delay(5, d1))
+        # d2
+        reference += instructions.Delay(11, d2)
+        self.assertEqual(schedule, reference)
+
+    def test_align_right(self):
+        """Test the right alignment context."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+        d2 = pulse.DriveChannel(2)
+
+        with pulse.build() as schedule:
+            with pulse.align_right():
+                pulse.delay(11, d2)
+                pulse.delay(3, d0)
+                with pulse.align_right():
+                    pulse.delay(5, d1)
+                    pulse.delay(7, d0)
+
+        reference = pulse.Schedule()
+        # d0
+        reference = reference.insert(1, instructions.Delay(3, d0))
+        reference = reference.insert(4, instructions.Delay(7, d0))
+        # d1
+        reference = reference.insert(6, instructions.Delay(5, d1))
+        # d2
+        reference += instructions.Delay(11, d2)
+
+        self.assertEqual(schedule, reference)
+
+    def test_group(self):
```

Comment:
This still doesn't really show me what this is. In practice, it's the same as align left? Is there a test that would distinguish their behavior?

Reply:
The idea was this would allow you to group a set of instructions with relative timing and also force the compilation of any lazy circuits without knowing the default scheduling context. However, with the current implementation this isn't trivial as it requires knowledge of the context stack (I want it to be with a future revision). So I am removing `group` for now and will think about reading. 

Removed in a2cda9e29.